litr/src/main/java/com/linkedin/android/litr/io/MediaMuxerMediaTarget.java

```diff
@@ -40,16 +45,51 @@
 
     private MediaFormat[] mediaFormatsToAdd;
 
+    private ParcelFileDescriptor parcelFileDescriptor;
     private String outputFilePath;
     private int numberOfTracksToAdd;
     private int trackCount;
 
+    /**
+     * This constructor to support scoped-storage enforcement in Android 10+, by using Uri instead of directly writing to File
+     */
+    public MediaMuxerMediaTarget(@NonNull Context context, @NonNull Uri outputFileUri,
```

Comment:
I like the idea of this new constructor. But I feel its logic might be overcomplicated. What it should do is this:
 - if API level is >= 26 it should use FileDescriptor when creating a MediaMuxer
 - at lower API levels it should behave exactly the way current constructor does - accept only "file://"  URI
 - otherwise (non-file URI in API level < 26) we should throw a `MediaTargetException`, maybe add a new error type in it for UNSUPPORTED_URI_TYPE with "URI type not supported at API level below 26" description.
This way, a constructor can create a `MediaMuxer` and pass it to `init` method to be configured.

Reply:
Fixed